### PR TITLE
Unify SpanHelper class

### PR DESCRIPTION
**Pull request by Kielek**
Follow up to #1595
Propagate changes from #1577 to AspNet Instrumentation package

## Changes

* Unify `SpanHelper` class.
* `SpanHelper` operates on native `ActivityStatusCode` instead of specific `Tracer.Status` (it was done a long time ago in main repository).
* Drop IncludeSharedSpanHelper

For significant contributions please make sure you have completed the following items:

* ~~[ ] Appropriate `CHANGELOG.md` updated for non-trivial changes~~
* ~~[ ] Design discussion issue #~~
* ~~[ ] Changes in public API reviewed~~

### CODE REVIEW

File: src/Shared/SpanHelper.cs [hunk 1]
```diff
@@ -16,37 +16,37 @@ internal static class SpanHelper
     /// </summary>
     /// <param name="httpStatusCode">Http status code.</param>
     /// <returns>Resolved span <see cref="Status"/> for the Http status code.</returns>
-    public static Status ResolveSpanStatusForHttpStatusCode(int httpStatusCode)
+    public static ActivityStatusCode ResolveSpanStatusForHttpStatusCode(int httpStatusCode)
```
Comment (vishweshbankwar) [inline src/Shared/SpanHelper.cs:19]: Do we need this one?.

Comment (Kielek) [reply src/Shared/SpanHelper.cs:19]: It is used by 2 instrumentation packages. OWIN and one more.

Comment (vishweshbankwar) [reply src/Shared/SpanHelper.cs:19]: Got it - we should update those to use the one below.

File: src/OpenTelemetry.Instrumentation.Owin/CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,8 @@
   ([#1335](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1335))
 * Fix description for `http.server.request.duration` metric.
   ([#1538](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1538))
+* Span status is calucated based on [semantic convention for server spans](https://github.com/open-telemetry/semantic-conventions/blob/v1.24.0/docs/http/http-spans.md#status).
```
Comment (vishweshbankwar) [inline src/OpenTelemetry.Instrumentation.Owin/CHANGELOG.md:13]:
```suggestion
* Span status is set based on [semantic convention for server spans](https://github.com/open-telemetry/semantic-conventions/blob/v1.24.0/docs/http/http-spans.md#status).
```

**Review by vishweshbankwar — APPROVED** [review]

**Pull request merged**